New Settings UI: Disable settings in send-only countries (4027)

hmouhtar opened the pull request:
### Description

This PR updates the settings screen to hide all settings if the store's current country is classified as a send-only country. Instead, a message explaining this limitation will be displayed.

### Steps to Test

1. Enable the new settings UI
2. Set the WooCommerce store country to one that is within the send-only list. This list can be found in [PCP-3970](https://inpsyde.atlassian.net/browse/PCP-3790).
3. Verify that no settings appear on the PayPal settings screen. Instead, a message explaining this limitation should be displayed.

### Screenshots

![WooCommerce-settings-‹-WooCommerce-PayPal-Payments-—-WordPress-12-19-2024_02_41_PM](https://github.com/user-attachments/assets/1b72c06d-2860-4e93-b177-7ae778dd5488)

hmouhtar committed:
New Settings UI: Disable settings in send-only countries

hmouhtar commented:
@stracker-phil I wasn’t sure if there is a specific style we need to use for notices, so I haven’t added any styling for the notice yet, just plain text.

stracker-phil commented on modules/ppcp-settings/resources/js/Components/Screens/SendOnlyMessage.js:
We should translate this message via `__()` or inject the string from server-side using `wp_localize_script()`

hmouhtar replied:
Hi @stracker-phil, I added localization to the message in my latest commit 👍🏻

stracker-phil reviewed (changes requested):
Localization is missing 😉

hmouhtar committed:
New Settings UI: Adjust send-only country message and notices

stracker-phil commented on modules/ppcp-paylater-configurator/src/PayLaterConfiguratorModule.php:
```diff
@@ -68,8 +69,9 @@ static function () use ( $c ) {
 				$current_page_id     = $c->get( 'wcgateway.current-ppcp-settings-page-id' );
 				$is_wc_settings_page = $c->get( 'wcgateway.is-wc-settings-page' );
 				$messaging_locations = $c->get( 'paylater-configurator.messaging-locations' );
+				$onboarding_state    = $c->get( 'onboarding.state' );
```
This uses the legacy onboarding status.

To check if the onboarding in the new UI is completed, you can use this logic:

```php
$onboarding_profile = $c->get( 'settings.data.onboarding' );
$onboarding_profile->get_completed(); // true = onboarding is done, false = still in the onboarding wizard
```

hmouhtar replied:
Hi @stracker-phil, I've switched to the new onboarding profile class on my latest commit 👍🏻

hmouhtar committed:
Switch to new onboarding status class

stracker-phil committed:
🔀 Merge branch 'trunk'

# Conflicts:
#	modules/ppcp-settings/src/Data/CommonSettings.php
#	modules/ppcp-settings/src/Endpoint/CommonRestEndpoint.php

stracker-phil committed:
💄 Make the send-only message look a bit nicer

stracker-phil committed:
✨ Improve the loading experience

Before this change, the settings screen was briefly visible, before the merchant details were loaded from the REST endpoint

stracker-phil committed:
♻️ Use a shorter attribute name for send-only

stracker-phil committed:
🚨 Fix phpcs warnings
